### ISSUE TITLE
Repeating fields?

### ISSUE THREAD

**Issue by wpcarro** [joaotavora/yasnippet#1021]
I'm trying to write a snippet for dictionaries in Python.

```
{
    "$1": $2,
    "$3": $4,
    "$5": $6,
}
```

Is it possible to continually tab and add more key-value pairs (e.g. `"$7": $8`, and so on)? I think this would be quite useful and create a structured editing feel.

**Comment by npostavs** [comment]
No, it isn't possible right now. How do you envision terminating the addition of key-value pairs, by the way?

**Comment by mohkale** [comment]
This seems like the kind of thing [auto-yasnippet](https://github.com/abo-abo/auto-yasnippet) was designed for. It works by creating single use snippets you can repeat multiple times.

In this case you'd have to write the following yourself (or create a snippet to expand to it):
```
{
"~key": ~value,
}
```

**NOTE**: not indented because python-mode auto indents on new lines and aya expands after point.

On the second line, run `aya-create` to create the snippet. This'll replace the `~` placeholders with their current values. You can delete the line once you've done this.

Then start writing and move to the indentation you want. Run `aya-expand` and it'll let you fill in the key and value fields as if you expanded a snippet. Once you're done it'll take you to the end of the line, hit return to move to the next line and repeat for all the entries you want to make.

**Comment by MirkoHernandez** [comment]
I was going to create an issue suggesting the same but I found this one. Here is how I'm currently implementing the repeating fields (an example from  prog-mode snippets). 

``` emacs-lisp
# -*- mode: snippet -*-
# name:parameters 
# key: p 
# type: command
# --
(yas-expand-snippet
(concat 
"$1 ,"
) nil nil
'((yas-after-exit-snippet-hook
'(lambda ()
	 (if (not current-prefix-arg)
	        (my/call-snippet "parameters")
	  (search-backward-regexp ",")
	 (delete-char 1)
	(fixup-whitespace)
)))))
```
The prefix argument is used to exit the snippet (control+u and then tab).  It would be nice to have a cleaner syntax for this particular use case, so the feature request would be an improved mechanism to achieve the same as the example code.

`my/call-snippet` is just a helper function.

``` emacs-lisp
(defun my/call-snippet (snippet)
  (yas-expand-snippet (yas-lookup-snippet  snippet)))
```
